src/components/Footer/Footer.js

```diff
@@ -63,23 +82,98 @@ const Footer = ({
+        const selectedUser = {
+          user: {
+            name: channelName,
+            image: item?.actor?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+          }
+        };
+        const members = [item?.actor?.id, profile.myProfile.user_id];
+        await createSignChat(members, selectedUser);
+      } else {
+        await sendMessageDM(item.id, 'post', 'SIGNED');
+      }
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDm: false});
+    }
+  };
+
+  const onPressDMAnon = async () => {
+    try {
+      setLoading({...loading, loadingDmAnon: true});
+      await sendMessageDM(item.id, 'post', 'ANONYMOUS');
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDmAnon: false});
+    }
+  };
+
+  const dataSheet = [
+    {
+      id: 1,
+      name: loading.loadingDm ? 'Loading...' : `Message ${username}`,
+      icon: <SendDMBlack />,
+      onPress: onPressDM
+    },
+    {
+      id: 2,
+      name: loading.loadingDmAnon ? 'Loading...' : 'Message using Incognito Mode',
+      icon: <SendDMAnonBlock />,
+      onPress: onPressDMAnon
+    }
+  ];
+
   return (
     <BlurredLayer toastOnly={true} isVisible={item?.isBlurredPost}>
       <View style={[styles.rowSpaceBetween, styles.container]}>
         <View style={styles.leftGroupContainer}>
-          <TouchableOpacity
-            testID="shareBtn"
-            disabled={item?.isBlurredPost}
-            style={styles.btn}
-            onPress={onPressShare}>
-            <View style={styles.btnShare}>
-              <MemoIc_share
-                color={item?.isBlurredPost ? COLORS.gray : undefined}
-                height={20}
-                width={21}
-              />
-            </View>
-          </TouchableOpacity>
+          {isShowDM && !isSelf ? (
+            <TouchableOpacity
+              testID="sendDM"
+              style={styles.btn}
+              onPress={() => refSheet.current.open()}>
+              <View
+                style={[
+                  styles.btnShare,
+                  {
+                    flexDirection: 'row',
+                    alignItems: 'center'
+                  }
+                ]}>
+                <MemoIc_senddm height={20} width={21} />
+                <Text
+                  style={[
+                    styles.text,
+                    {
+                      marginLeft: 4,
+                      fontWeight: '500'
+                    }
+                  ]}>
+                  DM
+                </Text>
+              </View>
+            </TouchableOpacity>
+          ) : (
+            <TouchableOpacity testID="shareBtn" style={styles.btn} onPress={onPressShare}>
+              <View style={styles.btnShare}>
+                <MemoIc_share height={20} width={21} />
+              </View>
+            </TouchableOpacity>
+          )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The rendering logic has been updated to include a new button for sending direct messages when the `isShowDM` prop is true and the user is not viewing their own post. The BottomSheetMenu is used to present the options for sending a direct message. No issues found here.
```diff
+ return (
+   <BlurredLayer toastOnly={true} isVisible={item?.isBlurredPost}>
+     <View style={[styles.rowSpaceBetween, styles.container]}>
+       <View style={styles.leftGroupContainer}>
+        {isShowDM && !isSelf ? (
+          <TouchableOpacity
+            testID="sendDM"
+            style={styles.btn}
+            onPress={() => refSheet.current.open()}>
+            <View
+              style={[
+                styles.btnShare,
+                {
+                  flexDirection: 'row',
+                  alignItems: 'center'
+                }
+              ]}>
+              <MemoIc_senddm height={20} width={21} />
+              <Text
+                style={[
+                  styles.text,
+                  {
+                    marginLeft: 4,
+                    fontWeight: '500'
+                  }
+                ]}>
+                DM
+              </Text>
+            </View>
+          </TouchableOpacity>
+        ) : (
+          <TouchableOpacity testID="shareBtn" style={styles.btn} onPress={onPressShare}>
+            <View style={styles.btnShare}>
+              <MemoIc_share height={20} width={21} />
+            </View>
+          </TouchableOpacity>
+        )}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->